ovirtlib4/system_service.py

```diff
@@ -198,6 +187,25 @@ def entity_type(self, entity_type):
         """ The entity type must be set by the inherit class """
         self._entity_type = entity_type
 
+
+class CollectionService(OvirtService):
+    """
+    Abstract class, represent an oVirt collection
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._entity_service = NotImplementedError
+
+    @property
+    def entity_service(self):
+        """ Return the service of an individual entity of the collection """
+        return self._entity_service
+
+    @entity_service.setter
+    def entity_service(self, entity_service):
+        """ The _entity_service must be set by the inherit class """
+        self._entity_service = entity_service
+
```

Comment:
Please see my notes regarding the use of builtin module [abc](https://docs.python.org/3.6/library/abc.html#module-abc)

Reply:
same ^